datalad/metadata/find_datasets.py

```diff
@@ -0,0 +1,117 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Interface for managing metadata
+"""
+
+__docformat__ = 'restructuredtext'
+
+
+import re
+from datalad.interface.base import Interface
+from datalad.distribution.dataset import datasetmethod, EnsureDataset, \
+    require_dataset
+from ..support.param import Parameter
+from ..support.constraints import EnsureNone
+from ..log import lgr
+from . import get_metadata, flatten_metadata_graph
+
+
+class FindDatasets(Interface):
+    """
+    """
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to perform the query operation on. If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory and/or the `path` given""",
+            constraints=EnsureDataset() | EnsureNone()),
+        match=Parameter(
+            args=("match",),
+            metavar='REGEX',
+            doc="expression to match against all meta data values"),
+        #match=Parameter(
+        #    args=('-m', '--match',),
+        #    metavar='REGEX',
+        #    action='append',
+        #    nargs=2,
+        #    doc="""Pair of two regular expressions to match a property and its
+        #    value.[CMD:  This option can be given multiple times CMD]"""),
+        report=Parameter(
+            args=('-r', '--report'),
+            metavar='PROPERTY',
+            action='append',
+            # could also be regex
+            doc="""name of the property to report for any match.[CMD:  This
+            option can be given multiple times. CMD] If none are given, all
+            properties are reported."""),
+    )
+
+    @staticmethod
+    @datasetmethod(name='query')
```

Comment:
I don't mind. As long as you don't mind that an option triggers using an almost completely disjoint implementation that likely operates on a different database than the cached metadata in the actual datasets.